### PR TITLE
Ugly ugly UI

### PR DESCRIPTION
**Pull request by n0izn0iz**
+ RegistryViewer
+ setContractDefaults
+ "Login" (lol..)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -8,40 +8,64 @@ var Production = loadContract("Production")*/
...
 	setRegistry(reg) {
 		this.setState({registry: reg})
 	}
+	setAddress(addr) {
+		this.setState({address: addr})
+	}
 	render() {
-		console.log("prodList: "+this.state.prodList)
 		var regStr = this.state.registry ? this.state.registry.address : "none"
+		var addrStr = this.state.address ? this.state.address : "none"
```
Comment (louptheron) [inline src/App.js]: let's name vars in a understandable way, like no addrStrSmthg or so, no IoT compiled file-size to optimise here ;)

File: src/components/RegistryViewer.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+// Bob: 0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1
+// Alice: 0xffcf8fdee72ac11b5c542428b35eef5769c409f0
+
+
+function applyConstraints(Production, prodAddr, constraints, from)
+{
+	return (Production.at(prodAddr).then((prod) => {
+		return (prod.buyer().then((prodBuyer) => {
+			if ((constraints.buyer && prodBuyer === from) ||
+					(!constraints.buyer && prodBuyer !== from))
```
Comment (louptheron) [inline src/components/RegistryViewer.js]: If there is not constraint on buyer, we can return it no ? why check `prodBuyer !== from` ?

Comment (n0izn0iz) [reply src/components/RegistryViewer.js]: cause we don't want to see our own offers, no? if we use the same address for buying and selling

Comment (n0izn0iz) [reply src/components/RegistryViewer.js]: if you want to see all offers, just remove the constraints prop or the buyer element

Comment (louptheron) [reply src/components/RegistryViewer.js]: ok, so the `applyConstraints` function only check for my offers for now (or all offers), not clear when reading the code

Comment (n0izn0iz) [reply src/components/RegistryViewer.js]: it  returns:
+ my offers if buyer is true
+ others offers if buyer if false
+ all offers if there is no constraints

Comment (louptheron) [reply src/components/RegistryViewer.js]: Yep, I was just saying that I think this is cleaner to know what a function is doing by reading a function or variable's name, and not having to go to the top function's caller or reading the implementation :)

**Review by louptheron — APPROVED** [review]

**Pull request merged**